DevCycle/ObjC/ObjCDVCClient.swift

```diff
@@ -6,13 +6,74 @@
 
 import Foundation
 
+@objc(DVCClientBuilder)
+public class ObjCClientBuilder: NSObject {
+    @objc public var environmentKey: String?
+    @objc public var user: ObjCDVCUser?
+    @objc public var options: ObjCDVCOptions?
+}
+
 @objc(DVCClient)
 public class ObjCDVCClient: NSObject {
-    
     var client: DVCClient?
-    
     @objc public var eventQueue: [ObjCDVCEvent] = []
     
+    @objc(build:block:onInitialized:)
+    public static func build(
+        block: ((ObjCClientBuilder) -> Void),
+        onInitialized: ((Error?) -> Void)?
+    ) throws -> ObjCDVCClient {
+        let builder = ObjCClientBuilder()
+        block(builder)
+        let client = try ObjCDVCClient(builder: builder, onInitialized: onInitialized)
+        return client
+    }
+    
+    @objc(initialize:user:err:)
+    public static func initialize(
+        environmentKey: String,
+        user: ObjCUserBuilder
+    ) throws -> ObjCDVCClient {
+        try self.initialize(
+            environmentKey: environmentKey,
+            user: user,
+            options: nil,
+            onInitialized: nil
+        )
+    }
+        
+    @objc(initialize:user:options:err:)
+    public static func initialize(
+        environmentKey: String,
+        user: ObjCUserBuilder,
+        options: ObjCOptionsBuilder?
+    ) throws -> ObjCDVCClient {
+        try self.initialize(
+            environmentKey: environmentKey,
+            user: user,
+            options: options,
+            onInitialized: nil
+        )
+    }
+        
+    @objc(initialize:user:options:err:onInitialized:)
+    public static func initialize(
+        environmentKey: String,
+        user: ObjCUserBuilder,
```

Comment:
why are we passing in a builder class here and not an instance of the user that's been built?

Reply:
As I described in the description the `initialize:` does the building so we only need one error object.

Reply:
If we went with this method I would probably just remove the `Builder` part of these names.